src/main/scala/nl/knaw/dans/lib/encode/package.scala

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (C) 2016 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.lib
+
+import com.google.common.net.PercentEscaper
+import java.nio.file.Path
+import scala.collection.JavaConverters._
+
+package object encode {
+
+  // PercentEscaper that escapes all characters except '_' and English alphanumerical characters
+  private val bagStorePercentEscaper = new PercentEscaper("_", false)
```

Comment:
How many times is this object created when you call it from different locations in the same project (preferably test this with using it in multiple files)? I hope it's once, but I'm not certain about the behavior of a `package object`. @jo-pol do you have experience with this?

Reply:
Never gave it a second thought. Can imagine it might be once per class-loader, whatever that implicates. Is this operation so expensive, and how could you avoid it?

Reply:
@rvanheest: Doesn't seem to be a very heavy type of object. Why are you worried about its being created often?

Reply:
tried it and it is only created once in the way it is written right now... So that makes me happy 😃

@janvanmansum have you looked at the implementation of the constructor. It does (1) a null check, (2) a pattern match, (3) string concatenation, (4) character search and (5) 2 more for-loops to create 'save octets'. No wonder Guava has a singleton class with default implementations! It's all about performance.

Reply:
@rvanheest : I have looked at the constructor. As I didn't see anything that I'd expect to have a significant performance impact (let's say file or network access) in my estimation you worry was really not warranted. Anyway, the issue moot.